shell/server/pre-meteor.js

```diff
@@ -471,6 +490,12 @@ Meteor.startup(() => {
         return;
       }
 
+      if (id.match(/^selftest/)) {
```

Comment:
Technically, a valid grain host could begin with "selftest". Note that "static" is not a valid grain host because grain host IDs are always longer than that, but for "selftest" you're only matching a prefix. I think it would make sense to permit hyphens in wildcard IDs so that you can make this "selftest-*", which is never a valid grain host.